File: src/schemas/widget.ts
import Topic from '../interfaces/topic';
import { Schema, model, Types } from 'mongoose';

export type Widget = {
  _id: Types.ObjectId;
  displayName: string;
  userId: string;
  enabled: boolean;
  token: string;
  template: string;
  templateId: string;
  templateRaw: string;
  settings: string;
  scopes: Topic[];
};

export const WidgetModel = model(
  'Widget',
  new Schema({
    displayName: String,
    userId: String,
    enabled: Boolean,
    token: String,
    template: String,
    templateId: String,
    templateRaw: String,
    settings: String,
    scopes: [String],
  }),
);

File: src/main/java/com/staroverlay/eventsub/entities/Widget.java
package com.staroverlay.eventsub.entities;

import com.sammwy.classserializer.annotations.Prop;
import com.sammwy.milkshake.Entity;

public class Widget extends Entity {
  @Prop
  public String displayName;

  @Prop
  public String userId;

  @Prop
  public boolean enabled;

  @Prop
  public String token;

  @Prop
  public String templateId;

  @Prop
  public String templateRaw;

  @Prop
  public String settings;

  @Prop
  public String[] scopes;
}
